src/poliastro/twobody/propagation.py

```diff
@@ -462,12 +462,6 @@ def propagate(orbit, time_of_flight, *, method=farnocchia, rtol=1e-10, **kwargs)
     else:
         pass
 
-    # Use the highest precision we can afford
-    # np.atleast_1d does not work directly on TimeDelta objects
-    jd1 = np.atleast_1d(time_of_flight.jd1)
-    jd2 = np.atleast_1d(time_of_flight.jd2)
-    time_of_flight = time.TimeDelta(jd1, jd2, format="jd", scale=time_of_flight.scale)
-
     rr, vv = method(
         orbit.attractor.k, orbit.r, orbit.v, time_of_flight.to(u.s), rtol=rtol, **kwargs
```

Comment:
Most errors in the propagators are of this nature:

```
>       results = [farnocchia_fast(k, r0, v0, tof) for tof in tofs]
E       TypeError: 'numpy.float64' object is not iterable
```

As it was written in the comments, "np.atleast_1d does not work directly on TimeDelta objects", but `time_of_flight.reshape(1).to(u.s)` does work!

Reply:
In fact, this diff makes all current tests pass:

```diff
diff --git a/src/poliastro/twobody/propagation.py b/src/poliastro/twobody/propagation.py
index c709d68b..d8ed8e36 100644
--- a/src/poliastro/twobody/propagation.py
+++ b/src/poliastro/twobody/propagation.py
@@ -463,7 +463,7 @@ def propagate(orbit, time_of_flight, *, method=farnocchia, rtol=1e-10, **kwargs)
         pass
 
     rr, vv = method(
-        orbit.attractor.k, orbit.r, orbit.v, time_of_flight.to(u.s), rtol=rtol, **kwargs
+        orbit.attractor.k, orbit.r, orbit.v, time_of_flight.reshape(-1).to(u.s), rtol=rtol, **kwargs
     )
 
     # TODO: Turn these into unit tests
```

...but the original problem #901 stands! Which means that there's some buggy behavior not covered by the tests.